apps/mobile/src/components/Dropdown/index.tsx

```diff
@@ -0,0 +1,173 @@
+import {useCallback, useMemo, useState} from 'react'
+import {TouchableOpacity, TouchableWithoutFeedback} from 'react-native'
+import {getTokens, Stack, styled, Text} from 'tamagui'
+import Image from '../Image'
+import chevronDownSvg from './images/chevronDownSvg'
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+} from 'react-native-reanimated'
+
+type Size = 'small' | 'large'
+
+export interface RowProps<T> {
+  type: T
+  title: string
+}
+
+interface Props<T> {
+  activeRowType: T
+  setActiveRowType: (rowType: T) => void
+  rows: Array<RowProps<T>>
+  size: Size
+}
+
+interface DropdownCellProps {
+  isOpen: boolean
+  isLast: boolean
+  onPress: () => void
+  title: string
+  size: Size
+}
+
+const StyledRow = styled(Stack, {
+  fd: 'row',
+  ai: 'center',
+  jc: 'space-between',
+  px: '$5',
+  py: '$4',
+  variants: {
+    isLast: {
+      true: {
+        borderBottomRightRadius: '$5',
+        borderBottomLeftRadius: '$5',
+      },
+    },
+    isOpen: {
+      true: {
+        bc: '$grey',
+      },
+      false: {
+        bc: '$primary',
+      },
+    },
+  },
+})
+
+function DropdownCell({
+  title,
+  isLast,
+  isOpen,
+  onPress,
+  size,
+}: DropdownCellProps): JSX.Element {
+  return (
+    <TouchableOpacity onPress={onPress}>
+      <StyledRow isOpen={isOpen} isLast={isLast}>
+        <Text
+          ff={size === 'small' ? '$body' : '$body600'}
+          fos={size === 'small' ? 16 : 18}
+          col={'$greyOnBlack'}
+        >
+          {title}
+        </Text>
+      </StyledRow>
+    </TouchableOpacity>
+  )
+}
+
+function Dropdown<T>({
+  activeRowType,
+  setActiveRowType,
+  rows,
+  size,
+}: Props<T>): JSX.Element {
+  const tokens = getTokens()
+  const [isOpen, setIsOpen] = useState<boolean>(false)
+  const activeRow = useMemo(
+    () => rows.find((row) => row.type === activeRowType),
+    [activeRowType, rows]
+  )
+
+  const rotation = useSharedValue(isOpen ? 180 : 0)
```

Comment:
I suggest not to use magic strings in the code
```
  const rotation = useSharedValue(isOpen ? Angle.OPEN : Angle.CLOSED)
```

Reply:
generally, I am for using constants instead of hard coded values. But in this case it's much more readable to have values directly instead of defined constants. What is open/closed angle? 

But nice catch. Thanks :)